WcaOnRails/app/jobs/submit_report_nag_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SubmitReportNagJob < ApplicationJob
+  queue_as :default
+
+  def nag_needed(competition)
+    (competition.report_nag_sent_at || competition.end_date) <= 8.days.ago
+  end
+
+  def perform
+   # Don't send nag emails for very old competitions without reports.
+   # Select only recent competitions or those that have already been nagged in the past.
+    Competition
+      .joins(:delegate_report)
+      .where("report_nag_sent_at IS NOT NULL OR start_date > ?", 1.month.ago)
```

Comment:
Let's just filter this to competitions that ended after June 1st, 2016. You should be able to share this code with https://github.com/jfly/worldcubeassociation.org/blob/aaf76de145e752b873aa9fef65ca53333c62b8c7/WcaOnRails/app/controllers/delegate_reports_controller.rb#L38-L45.